snapcraft/storeapi/__init__.py

```diff
@@ -89,6 +89,22 @@ def _get_name_from_snap_file(snap_path):
     return snap_yaml['name']
 
 
+def _macaroon_auth(conf):
+    """Format a macaroon and its associated discharge.
+
+    :return: A string suitable to use in an Authorization header.
+
+    """
+    root_macaroon_raw = conf.get('macaroon')
+    unbound_raw = conf.get('unbound_discharge')
+    root_macaroon = macaroons.Macaroon.deserialize(root_macaroon_raw)
+    unbound = macaroons.Macaroon.deserialize(unbound_raw)
+    bound = root_macaroon.prepare_for_request(unbound)
+    auth = 'Macaroon root={}, discharge={}'.format(root_macaroon_raw,
+                                                   bound.serialize())
```

Comment:
I don't believe we ever disable asserts in Launchpad production, but I take your point.  I'll fix the issues you've raised; thanks!

A deserialize check is probably good enough in practice, although personally I think it's still worth a low-level paranoia check on specific allowed characters when embedding it in an Authorization header, but perhaps that's just me.